packages/backend/src/index.ts

```diff
@@ -113,12 +113,28 @@ type AddPlugin = {
   isOptional?: false;
 } & AddPluginBase;
 
+type OptionalPluginOptions = {
+  key?: string;
+  path?: string;
+};
+
 type AddOptionalPlugin = {
   isOptional: true;
   config: Config;
-  options?: { key?: string; path?: string };
+  options?: OptionalPluginOptions;
 } & AddPluginBase;
 
+const optionalDynamicPlugins: { [key: string]: OptionalPluginOptions } = {
+  techdocs: {},
+  argocd: {},
+  sonarqube: {},
+  kubernetes: {},
+  'azure-devops': { key: 'enabled.azureDevOps' },
+  jenkins: {},
+  ocm: {},
+  gitlab: {},
+};
```

Comment:
```suggestion
const OPTIONAL_DYNAMIC_PLGUINS = {
  techdocs: {},
  argocd: {},
  sonarqube: {},
  kubernetes: {},
  'azure-devops': { key: 'enabled.azureDevOps' },
  jenkins: {},
  ocm: {},
  gitlab: {},
} as const satisfies { [key: string]: OptionalPluginOptions };
```

Please use MACRO_CASE and the `as const` assertion when specifying constants 🙂